db_stress_tool/db_stress_common.cc

```diff
@@ -16,6 +16,7 @@
 #include "util/file_checksum_helper.h"
 #include "util/xxhash.h"
 
+ROCKSDB_NAMESPACE::Env* db_stress_listener_env = nullptr;
```

Comment:
Can `db_stress_env` be used?

Reply:
Yes, for regular case,  but when fault_injection is enabled, the raw_env is set to the FaultInjectionTestFs(), and so are the db_stress_env and env_wrapper_guard. This results in faultinjection filesystem to be used every where. 
But earlier, I have got feedback that it is better not to  use the the fault injection filesystem in DbStressListener. 
(I had sent a previous iteration of this patch using db_stress_env).
So current approach does not let DbStressListener to use the fault injection FS.

Reply:
I see. Thanks for providing the context.

Reply:
`DbStressListener` uses the `FileSystem` ptr to access a `unique_ids` file. What would be ideal is adding support to `FaultInjectionFS` so that we can control whether we inject errors to certain file types.

Reply:
Ok, thanks for the suggestions, I will look into this. 
Earlier PR I sent was #9175 ,  one suggestion by [pdillinger](https://github.com/facebook/rocksdb/pull/9175#issuecomment-973046843) was to copy the raw_env before it was wrapped by the FaultFS(this current version does that). Could you please take a look at that also. Thanks.

Reply:
@riversand963 @pdillinger I looked at modifying FaultInjectionTestFS to not inject errors to specific file types, but I am a bit not convinced as the .unique_ids file does not have any type attributed to it,  could you please add more context on how you want it done. 
My current implementation is based on @pdillinger feedback in PR #9175.
db_stress is broken for us, would appreciate if you can take a look into this. Thanks.

Reply:
We can filter by file type and name. I am thinking about something similar to the following in `FaultInjectionTestFS`/`FaultInjectionTestEnv`.
```
if (file name fully matches) {
  return;  // skip injecting errors
} else if (file type falls into one of the specified types) {
  return; // skip injecting errors
}
```

Reply:
I prefer this approach of accessing the right abstraction layer rather than complicating FaultInjectionTestFS with specific exceptions.

Reply:
I think having a single `env` with **centralized** knowledge of what types of file to skip for fault injection is simpler and more scalable than having multiple `env`s around. In this case, we need to skip "unique-ids" for listeners, so we add a new `stress_listener_env`. In the future, what if we need to skip fault injection for another file or type of files?